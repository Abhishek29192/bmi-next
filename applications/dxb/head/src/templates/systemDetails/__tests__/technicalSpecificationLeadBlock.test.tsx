import React from "react";
import { render } from "@testing-library/react";
import Component from "../technicalSpecificationLeadBlock";
import {
  Classification,
  ClassificationCodeEnum,
  Feature
} from "../../../components/types/pim";
import "@testing-library/jest-dom";

const technicalSpecClassifications: Classification[] = [
  {
    code: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
    name: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
    features: [
      {
        name: "feature1",
        code: "bmiSystemsClassificationCatalog/1.0/systemAttributes.roofbuildup",
        featureValues: [{ value: "200" }]
      }
    ]
  }
];

const featureUnit = {
  name: "height",
  unitType: "metric",
  symbol: "in"
};

const featureWithUnit: Feature = {
  ...technicalSpecClassifications[0].features[0],
  featureUnit
};

describe("TechnicalSpecificationLeadBlock tests", () => {
  beforeEach(() => {
    // resolve useDimensions (useState) hook in ProductFeatureTable
    jest.mock("react", () => ({
      ...(jest.requireActual("react") as any),
      useState: (initial) => [initial, jest.fn()]
    }));
    jest
      .spyOn(window, "requestAnimationFrame")
      .mockImplementation((callback: FrameRequestCallback): number => {
        callback(0);
        return 0;
      });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  describe("should render correctly", () => {
    describe("when One classifications provided", () => {
      it("With no feature units", () => {
        const { container, queryByText } = render(
          <Component
            technicalSpecClassifications={technicalSpecClassifications}
          />
        );
        const accordion = container.querySelectorAll(".Accordion");
        const techTable = container.querySelectorAll(
          ".SystemDetailsTechnicalSpec"
        );
        const unitTypeText = queryByText(featureUnit.symbol, { exact: false });

        expect(container).toMatchSnapshot();
        expect(accordion.length).toBeFalsy();
        expect(techTable.length).toBeTruthy();
        expect(unitTypeText).not.toBeInTheDocument();
      });

      it("With a feature units", () => {
        const classifications: Classification[] = [
          {
            name: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            code: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            features: [featureWithUnit]
          }
        ];
        const { container, queryByText } = render(
          <Component technicalSpecClassifications={classifications} />
        );
        const unitTypeText = queryByText(featureUnit.symbol, { exact: false });

        expect(container).toMatchSnapshot();
        expect(unitTypeText).toBeInTheDocument();
      });
    });

    describe("when Multiple classifications provided", () => {
      it("With no feature units", () => {
        const { container, queryByText } = render(
          <Component
            technicalSpecClassifications={[
              ...technicalSpecClassifications,
              ...technicalSpecClassifications
            ]}
          />
        );
        const accordion = container.querySelectorAll(".Accordion");
        const techTable = container.querySelectorAll(
          ".SystemDetailsTechnicalSpec"
        );
        const unitTypeText = queryByText(featureUnit.symbol, { exact: false });

        expect(container).toMatchSnapshot();
        expect(accordion.length).toBeTruthy();
        expect(techTable.length).toBeTruthy();
        expect(unitTypeText).not.toBeInTheDocument();
      });

      it("With more than one feature units", () => {
        const classifications: Classification[] = [
          {
            name: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            code: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            features: [featureWithUnit, featureWithUnit]
          },
          {
            name: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            code: ClassificationCodeEnum.SYSTEM_ATTRIBUTES,
            features: [featureWithUnit, featureWithUnit]
          }
        ];
        const { container, queryAllByText } = render(
          <Component technicalSpecClassifications={classifications} />
        );
        const unitTypeText = queryAllByText(featureUnit.symbol, {
          exact: false
        });

        expect(container).toMatchSnapshot();
        expect(unitTypeText.length).toBeTruthy();
      });
    });
  });
});
