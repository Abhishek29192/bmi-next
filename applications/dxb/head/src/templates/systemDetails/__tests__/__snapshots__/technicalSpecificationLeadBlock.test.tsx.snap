// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnicalSpecificationLeadBlock tests should render correctly when Multiple classifications provided With more than one feature units 1`] = `
<div>
  <div
    class="MuiGrid-root LeadBlock MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root LeadBlockContent MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
    >
      <div
        class="LeadBlockContentSection"
      >
        <div
          class="SystemDetailsTechnicalSpec"
        >
          <div
            class="accordionStyles-root-8 accordionStyles-no-inner-padding-11"
          >
            <div
              class="MuiPaper-root MuiAccordion-root accordionStyles-item-9 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                aria-disabled="false"
                aria-expanded="true"
                class="MuiButtonBase-root MuiAccordionSummary-root accordionStyles-summaryRoot-12 Mui-expanded accordionStyles-summaryExpanded-13"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content accordionStyles-summaryContent-14 Mui-expanded accordionStyles-summaryExpanded-13"
                >
                  <h6
                    class="MuiTypography-root Typography MuiTypography-h6"
                  >
                    systemAttributes
                  </h6>
                </div>
                <div
                  aria-disabled="false"
                  aria-hidden="true"
                  class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded accordionStyles-summaryExpanded-13 MuiIconButton-edgeEnd"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-entered"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <div
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root accordion-details"
                      >
                        <div>
                          <table
                            class="MuiTable-root Table Table--row-even-color Table--no-border"
                          >
                            <tbody
                              class="MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 in
                                </td>
                              </tr>
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 in
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiAccordion-root accordionStyles-item-9 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                aria-disabled="false"
                aria-expanded="true"
                class="MuiButtonBase-root MuiAccordionSummary-root accordionStyles-summaryRoot-12 Mui-expanded accordionStyles-summaryExpanded-13"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content accordionStyles-summaryContent-14 Mui-expanded accordionStyles-summaryExpanded-13"
                >
                  <h6
                    class="MuiTypography-root Typography MuiTypography-h6"
                  >
                    systemAttributes
                  </h6>
                </div>
                <div
                  aria-disabled="false"
                  aria-hidden="true"
                  class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded accordionStyles-summaryExpanded-13 MuiIconButton-edgeEnd"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-entered"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <div
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root accordion-details"
                      >
                        <div>
                          <table
                            class="MuiTable-root Table Table--row-even-color Table--no-border"
                          >
                            <tbody
                              class="MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 in
                                </td>
                              </tr>
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 in
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TechnicalSpecificationLeadBlock tests should render correctly when Multiple classifications provided With no feature units 1`] = `
<div>
  <div
    class="MuiGrid-root LeadBlock MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root LeadBlockContent MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
    >
      <div
        class="LeadBlockContentSection"
      >
        <div
          class="SystemDetailsTechnicalSpec"
        >
          <div
            class="accordionStyles-root-1 accordionStyles-no-inner-padding-4"
          >
            <div
              class="MuiPaper-root MuiAccordion-root accordionStyles-item-2 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                aria-disabled="false"
                aria-expanded="true"
                class="MuiButtonBase-root MuiAccordionSummary-root accordionStyles-summaryRoot-5 Mui-expanded accordionStyles-summaryExpanded-6"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content accordionStyles-summaryContent-7 Mui-expanded accordionStyles-summaryExpanded-6"
                >
                  <h6
                    class="MuiTypography-root Typography MuiTypography-h6"
                  >
                    systemAttributes
                  </h6>
                </div>
                <div
                  aria-disabled="false"
                  aria-hidden="true"
                  class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded accordionStyles-summaryExpanded-6 MuiIconButton-edgeEnd"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-entered"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <div
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root accordion-details"
                      >
                        <div>
                          <table
                            class="MuiTable-root Table Table--row-even-color Table--no-border"
                          >
                            <tbody
                              class="MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiPaper-root MuiAccordion-root accordionStyles-item-2 Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                aria-disabled="false"
                aria-expanded="true"
                class="MuiButtonBase-root MuiAccordionSummary-root accordionStyles-summaryRoot-5 Mui-expanded accordionStyles-summaryExpanded-6"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiAccordionSummary-content accordionStyles-summaryContent-7 Mui-expanded accordionStyles-summaryExpanded-6"
                >
                  <h6
                    class="MuiTypography-root Typography MuiTypography-h6"
                  >
                    systemAttributes
                  </h6>
                </div>
                <div
                  aria-disabled="false"
                  aria-hidden="true"
                  class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded accordionStyles-summaryExpanded-6 MuiIconButton-edgeEnd"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-entered"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <div
                      role="region"
                    >
                      <div
                        class="MuiAccordionDetails-root accordion-details"
                      >
                        <div>
                          <table
                            class="MuiTable-root Table Table--row-even-color Table--no-border"
                          >
                            <tbody
                              class="MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root"
                              >
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                                >
                                  feature1
                                </td>
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                                >
                                  200 
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TechnicalSpecificationLeadBlock tests should render correctly when One classifications provided With a feature units 1`] = `
<div>
  <div
    class="MuiGrid-root LeadBlock MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root LeadBlockContent MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
    >
      <div
        class="LeadBlockContentSection"
      >
        <div
          class="SystemDetailsTechnicalSpec"
        >
          <div>
            <table
              class="MuiTable-root Table Table--row-even-color"
            >
              <thead
                class="MuiTableHead-root ColorPair ColorPair--pearl"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    systemAttributes
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  >
                    feature1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    200 in
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TechnicalSpecificationLeadBlock tests should render correctly when One classifications provided With no feature units 1`] = `
<div>
  <div
    class="MuiGrid-root LeadBlock MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      class="MuiGrid-root LeadBlockContent MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-8"
    >
      <div
        class="LeadBlockContentSection"
      >
        <div
          class="SystemDetailsTechnicalSpec"
        >
          <div>
            <table
              class="MuiTable-root Table Table--row-even-color"
            >
              <thead
                class="MuiTableHead-root ColorPair ColorPair--pearl"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  >
                    systemAttributes
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head"
                    scope="col"
                  />
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
                  >
                    feature1
                  </td>
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
                  >
                    200 
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
